src/ray/gcs/gcs_server/node_info_handler_impl.h

```diff
@@ -57,8 +61,32 @@ class DefaultNodeInfoHandler : public rpc::NodeInfoHandler {
                              SendReplyCallback send_reply_callback) override;
 
  private:
+  void UnregisterNode(const ClientID &node_id, rpc::GcsNodeInfo &node_info,
+                      rpc::UnregisterNodeReply *reply,
+                      const SendReplyCallback &send_reply_callback);
+
+  void DeleteResources(const ClientID &node_id,
+                       const std::vector<std::string> &resource_names,
+                       const gcs::NodeInfoAccessor::ResourceMap &delete_resources,
+                       DeleteResourcesReply *reply,
+                       const SendReplyCallback &send_reply_callback);
+
   gcs::RedisGcsClient &gcs_client_;
   gcs::GcsNodeManager &gcs_node_manager_;
+  gcs::GcsPubSub gcs_pub_;
+  const std::string node_channel_ = TablePubsub_Name(TablePubsub::CLIENT_PUBSUB);
+  const std::string node_resource_channel_ =
+      TablePubsub_Name(TablePubsub::NODE_RESOURCE_PUBSUB);
+
+  /// Mutex to protect the nodes_cache_ field.
+  absl::Mutex node_mutex_;
+  /// A mapping from node id to node info.
+  std::unordered_map<ClientID, rpc::GcsNodeInfo> nodes_cache_ GUARDED_BY(node_mutex_);
```

Comment:
I think we should put this in `GcsNodeManager`.

Reply:
fixed